integration-cli/daemon/daemon.go

```diff
@@ -757,6 +758,16 @@ func (d *Daemon) ReloadConfig() error {
 	return nil
 }
 
+// NewClient creates new client based on daemon's socket path
+func (d *Daemon) NewClient() (*client.Client, error) {
+	var httpClient *http.Client
+	cli, err := client.NewClient(d.Sock(), "", httpClient, nil)
```

Comment:
The problem is that the client is not initialised with a version, causing the version check to produce an error. There's a couple of solutions;

1. Initialize the client with the default version supported by this client;

```patch
diff --git a/integration-cli/daemon/daemon.go b/integration-cli/daemon/daemon.go
index 6c14136..d4da3d3 100644
--- a/integration-cli/daemon/daemon.go
+++ b/integration-cli/daemon/daemon.go
@@ -14,6 +14,7 @@ import (
        "strings"
        "time"

+       "github.com/docker/docker/api"
        "github.com/docker/docker/api/types/events"
        "github.com/docker/docker/client"
        "github.com/docker/docker/integration-cli/checker"
@@ -761,7 +762,7 @@ func (d *Daemon) ReloadConfig() error {
 // NewClient creates new client based on daemon's socket path
 func (d *Daemon) NewClient() (*client.Client, error) {
        var httpClient *http.Client
-       cli, err := client.NewClient(d.Sock(), "", httpClient, nil)
+       cli, err := client.NewClient(d.Sock(), api.DefaultVersion, httpClient, nil)
        if err != nil {
                return nil, err
        }
```

2. Update the version-check to assume "latest supported version" if no version is set

```patch
diff --git a/client/errors.go b/client/errors.go
index e0effaf..b8905df 100644
--- a/client/errors.go
+++ b/client/errors.go
@@ -3,6 +3,7 @@ package client
 import (
        "fmt"

+       "github.com/docker/docker/api"
        "github.com/docker/docker/api/types/versions"
        "github.com/pkg/errors"
 )
@@ -228,7 +229,11 @@ func IsErrPluginPermissionDenied(err error) bool {
 // NewVersionError returns an error if the APIVersion required
 // if less than the current supported version
 func (cli *Client) NewVersionError(APIrequired, feature string) error {
-       if versions.LessThan(cli.version, APIrequired) {
+       version := cli.version
+       if version == "" {
+               version = api.DefaultVersion
+       }
+       if versions.LessThan(version, APIrequired) {
                return fmt.Errorf("%q requires API version %s, but the Docker daemon API version is %s", feature, APIrequired, cli.version)
        }
        return nil
```

3. Allow version-negotiation if the version that's set for the client is empty

```patch
diff --git a/client/client.go b/client/client.go
index 222aa50..62a3db3 100644
--- a/client/client.go
+++ b/client/client.go
@@ -260,8 +260,13 @@ func (cli *Client) NegotiateAPIVersionPing(p types.Ping) {
                p.APIVersion = "1.24"
        }

-       // if server version is lower than the current cli, downgrade
-       if versions.LessThan(p.APIVersion, cli.ClientVersion()) {
+       // if the client is not initialized with a version, start with the latest supported version
+       if cli.version == "" {
+               cli.version = api.DefaultVersion
+       }
+
+       // if server version is lower than the maximum version supported by the Client, downgrade
+       if versions.LessThan(p.APIVersion, api.DefaultVersion) {
                cli.version = p.APIVersion
        }
 }
```

(3. also would require doing version negotiation after the client is initialised)

I'll open pull requests to discuss 2. and 3., but doing 1. would resolve the test failures